### ISSUE TITLE
Voice - Audio Stream/Speech Text to Talk

### ISSUE THREAD

**Issue by Pugalit** [Nexmo/nexmo-dotnet#149]
I am using .net SDK Client in ASP.Net Web API Project.
**Question 1:** 
I want to play the audio stream once the outbound call is answered by customer. After some other business logic, i want to stop playing the audio and conference one more customer in the same.

Option 1: 
In Call Answer Event, i am sending response http 200 response along with that i have included the content as Json Serialize object(action: stream, streamurl:test.mp3, loop=0)
Output: Audio is playing Infinity 
Issue: But i am not receiving any response from the Nexmo API for the playing AudioStream (result UUID to stop Audio play in future) 

Option 2: 
In Call Answer Event:
Using the below option i am trying to play the music in the outbound Answered call . 
var result = clientNexmo.Call.BeginStream(UUID, new Call.StreamCommand
{
stream_url = new[]
{
audiostreamURL
},
loop = 0
});
Post this method execution i am sending response http 200 response
Output: Audio is playing only 1 time/ Speech Text to Talk is not playing
Issue: Audio is playing only 1 time loop option is not working / Speech Text to Talk is not working also call disconnecting with in 6 to 7 seconds automatically.

**Question 2:** 
While Play an audio file or Play text to speech into a call, In which event i can track the Stream started, talk started, Stream completed or talk completed those details. I am not receiving theses details/events in Answer URL and Event URL.

### Which target framework(s) of .NET can you reproduce this issue against?
.NET Framework 4.6.2

### What operating system(s) is exhibiting this issue?
Window 2012 Server

### Version or commit hash
Nexmo.Api, Version=3.0.1.0,
Nexmo.Csharp.Client.3.2.4

**Comment by slorello89** [comment]
Hi @Pugalit - There's a bug in the command structures for stream and talk that is essentially ignoring the 0. The serializer in the VersionedApiRequest actually ignores default values - which for integers 0 is - I have opened #190 to address this issue and this should be available in the next release of SDK (4.3.1)

**Comment by slorello89** [comment]
Hi @Pugalit - this issue is fixed in the latest version of the SDK - version 4.3.1

FYI in order for you to use the loop parameter the way you'd like to, you will need to make sure that the NCCO you are using to create your call puts you in a conversation - if you were to, for example, just create the call with a talk action - when it reached the end of that action the call will end.

So try something like this if you want an initial talk action:

```csharp
            var talkAction = new TalkAction() { Text = "This is a text to speech call from Nexmo" };
            var conversationAction = new ConversationAction() { Name = "nexmo-conference-standard" };
            var ncco = new Ncco(talkAction, conversationAction);

            var results = client.Call.Do(new Call.CallCommand
            {
                to = new[]
                {
                    new Call.Endpoint {
                        type = "phone",
                        number = TO_NUMBER
                    }
                },
                from = new Call.Endpoint
                {
                    type = "phone",
                    number = NEXMO_NUMBER
                },

                NccoObj = ncco
            });
```

Prior to calling the Stream / talk action with your 0 loop parameter.

I'm closing this issue out - if you run into any problems feel free to reopen - let me know if you have any questions.

thanks!
